src/main/java/WebApp/Enterprise/Pollima/controller/CompanyController.java

```diff
@@ -23,26 +25,74 @@ public class CompanyController {
 
 
 
+    private MessageSource messageSource;
     private CompanyService companyService;
 
     @GetMapping
-    public String company(Model model){
+    public String company(Model model) {
         List<Company> companyList = companyService.findAll();
-        model.addAttribute("CompanyList",companyList);
-        model.addAttribute("AddNewCompany", CompanyForm.builder().build());
+        model.addAttribute("CompanyList", companyList);
+        model.addAttribute("saveNewCompany", CompanyForm.builder().build());
         return "company";
     }
-
+//
+//    String messageToShow = messageSource.getMessage("company.exist",null, LocaleContextHolder.getLocale());
+//    String companyName = form.getCompanyName();
+//    String theMessage = MessageFormat.format(messageToShow,companyName);
+//
+//        if (form.getID()==null) {
+//        companyService.findByCompanyName(form.getCompanyName()).ifPresent(company ->
+//                result.rejectValue("companyName", "error.company", theMessage)
+//        );
+//    }
     @PostMapping
-    private String saveCompany(@ModelAttribute("saveNewCompany") CompanyForm form){
+    private String saveCompany(@Valid @ModelAttribute("saveNewCompany") CompanyForm form, BindingResult result){
+        String messageToShow = messageSource.getMessage("company.exist",null, LocaleContextHolder.getLocale());
+        String companyName = form.getCompanyName();
+        String theMessage = MessageFormat.format(messageToShow,companyName);
+
+        if (form.getID()==null) {
+            companyService.findByCompanyName(form.getCompanyName()).ifPresent(company ->
+                    result.rejectValue("companyName", "error.company", theMessage)
```

Comment:
`error.company` does not exist in the messages.properties file.

Reply:
`result.rejectValue("companyName", "company.exist", new Object[]{form.getCompanyName()}, "company.exist")`